src/fastcs/backends/tango/dsr.py

```diff
@@ -0,0 +1,222 @@
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from types import MethodType
+from typing import Any
+
+import tango
+from tango import AttrWriteType, Database, DbDevInfo, DevState, server
+
+from fastcs.attributes import Attribute, AttrR, AttrRW, AttrW
+from fastcs.backend import (
+    _link_attribute_sender_class,
+    _link_single_controller_put_tasks,
+)
+from fastcs.controller import BaseController
+from fastcs.datatypes import Float
+from fastcs.mapping import Mapping
+
+
+@dataclass
+class TangoDSROptions:
+    dev_name: str = "MY/DEVICE/NAME"
+    dev_class: str = "FAST_CS_DEVICE"
+    dsr_instance: str = "MY_SERVER_INSTANCE"
+    debug: bool = False
+
+
+def _wrap_updater_fget(
+    attr_name: str, attribute: AttrR, controller: BaseController
+) -> Callable[[Any], Any]:
+    async def fget(self):
+        await attribute.updater.update(controller, attribute)
+        self.info_stream(f"called fget method: {attr_name}")
```

Comment:
I don't see where `info_stream` is defined. What is `self` in this context?

I was just trying to add some very simple tests to add some confidence that I don't break the tango backend going forward, but I am not sure what to pass to this function.

Reply:
`self` is a `Device` in this context. I have added some type hints and change the parameter name to show this. However these methods are defined dynamically, so static type hints do not work.